fractal_server/app/routes/admin/v2.py

```diff
@@ -307,3 +314,126 @@ async def flag_task_v1_as_v2_compatible(
     await db.close()
 
     return Response(status_code=status.HTTP_200_OK)
+
+
+class TaskV2Minimal(BaseModel):
+
+    id: int
+    name: str
+
+    type: str
+    command_non_parallel: Optional[str] = None
+    command_parallel: Optional[str] = None
+
+    source: str
+
+    meta_non_parallel: dict[str, Any] = Field(default_factory=dict)
+    meta_parallel: dict[str, Any] = Field(default_factory=dict)
+
+    owner: Optional[str]
+    version: Optional[str]
+
+    docs_info: Optional[str] = None
+    docs_link: Optional[HttpUrl] = None
+
+    input_types: dict[str, bool] = Field(default_factory=dict)
+    output_types: dict[str, bool] = Field(default_factory=dict)
+
+
+class ProjectUser(BaseModel):
+    id: int
+    email: EmailStr
+
+
+class TaskContext(BaseModel):
+
+    task_v2_minimal: TaskV2Minimal
+
+    workflow_id: Optional[int]
+    workflow_name: Optional[str]
+    project_id: Optional[int]
+    project_name: Optional[str]
+
+    project_users: Optional[list[ProjectUser]]
+
+
+@router_admin_v2.get("/task/", response_model=list[TaskContext])
+async def query_tasks(
+    id: Optional[int] = None,
+    source: Optional[str] = None,
+    version: Optional[str] = None,
+    name: Optional[str] = None,
+    owner: Optional[str] = None,
+    only_no_owner: bool = False,
+    max_number_of_results: int = 25,
+    user: User = Depends(current_active_superuser),
+    db: AsyncSession = Depends(get_async_db),
+) -> list[TaskContext]:
+
+    if (owner is not None) and (only_no_owner is True):
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail="Cannot query Tasks by owner if `only_no_owner` is True",
+        )
+
+    stm = select(TaskV2)
+
+    if id is not None:
+        stm = stm.where(TaskV2.id == id)
+    if source is not None:
+        stm = stm.where(TaskV2.source == source)
+    if version is not None:
+        stm = stm.where(TaskV2.version == version)
+    if name is not None:
+        stm = stm.where(TaskV2.name.icontains(name))
+
+    if owner is not None:
+        stm = stm.where(TaskV2.owner == owner)
+    elif only_no_owner is True:
+        stm = stm.where(TaskV2.owner == None)  # noqa E711
+
+    res = await db.execute(stm)
+    task_list = res.scalars().all()
+    if len(task_list) > max_number_of_results:
+        await db.close()
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=(
+                f"Too many Tasks ({len(task_list)} > {max_number_of_results})."
+                " Please add more query filters."
+            ),
+        )
+
+    contexts = []
+
+    for task in task_list:
+        stm = (
+            select(WorkflowV2)
+            .join(WorkflowTaskV2)
+            .where(WorkflowTaskV2.workflow_id == WorkflowV2.id)
+            .where(WorkflowTaskV2.task_id == task.id)
+        )
+        res = await db.execute(stm)
+        wf_list = res.scalars().all()
```

Comment:
What is the result, when no workflow is found? Is it like `None` or `[]`?
If it is `[]`, then we can avoid the `if wf_list` below.
If it is `None`, then let's turn it into if wf_list is not None` for clarity